src/TreeTest.cpp

```diff
@@ -48,3 +48,28 @@ TEST(Tree, GetPath) {
 	tree->getPath(path, tree->lastNode(), false);
 	ASSERT_EQ(true, path.size() > 1);
 }
+
+TEST(ASC, Tree) {
+	//test adaptive stepsize control
+	Tree<Vector2f> *tree = TreeFor2dPlane(
+		make_shared<GridStateSpace>(50, 50, 50, 50),
+		Vector2f(90, 90),	//	goal point
+		5);					//	step size
+
+	//	give it plenty of iterations so it's not likely to fail
+	const int maxIterations = 10000;
+	tree->setMaxIterations(maxIterations);
+	tree->setASCEnabled(true);
+
+	tree->setStartState(Vector2f(10, 10));
+	bool success = tree->run();	//	run with the given starting point
+
+	vector<Vector2f> path;
+	tree->getPath(path, tree->lastNode(), true);
+
+	bool varied = false;
+	for (int i = 0; i < path.size() - 2 && !varied; i++) {
+		if (path[i] != path[i + 1]) varied = true;
```

Comment:
How does this test work here?  It seems like it's just checking if successive points in the path are at the same location or not, so it should fail in all cases, right?